compound.py

```diff
@@ -137,72 +138,141 @@ def update_bot_order_volumes(
 ):
     """Update bot with new order volumes."""
 
+    bot_id = thebot["id"]
     bot_name = thebot["name"]
     base_order_volume = float(thebot["base_order_volume"])
     safety_order_volume = float(thebot["safety_order_volume"])
+    #### I know these are also down below, but to make calculation easier, I'm recalculating these
+    bo_profit = new_base_order_volume - base_order_volume
+    so_profit = new_safety_order_volume - safety_order_volume
+    #### Check from the database what the last stored values are
+    #### db_lastpassupdate = was the last update to BO executed or stored?
+    #### db_lastcalcbo = Last Calculated BO from Database
+    #### db_lastcalcso = Last Caclulated SO from Database
+    db_lastpassupdate = cursor.execute(
+        f"SELECT lastpassupdate FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastpassupdate value in db is {db_lastpassupdate}"
+    )
+    db_lastcalcbo = cursor.execute(
+        f"SELECT lastcalcbo FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastcalcbo value in db is {db_lastcalcbo}"
+    )
+    db_lastcalcso = cursor.execute(
+        f"SELECT lastcalcso FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastcalcso value in db is {db_lastcalcso}"
+    )
 
+    #### Check if the BO/SO was updated on the last pass
+    #### If not, add the value from the DB to the new BO
+    if db_lastpassupdate == 'No':
+        new_base_order_volume = (db_lastcalcbo + bo_profit)
+        new_safety_order_volume = (db_lastcalcso + so_profit)
```

Comment:
Iets om te overwegen. Nu krijgt deze functie een new_base_order_volume mee wat op de plek waar het wordt aangeroepen een bo + profit berekening is. Hier trek je dat weer uit elkaar wat logisch is voor wat je wilt doen. Wat als we op de plek waar we deze functie aanroepen enkel de profit meegeven? Dan kan je hier in de if de laatst opgeslagen bo + profit doen, en voegen we een else case toe waar je bot bo + profit doet. Dan zit het hele stukje van nieuwe bo berekenen bij elkaar en maakt het de lijn door de code wat makkelijker te volgen.

Ik hoop tenminste dat je mij nu volgt :P

Reply:
Ik heb dat inderdaad zitten bekijken en zag ook dat het wat dubbelop was...
Toen nog gekeken of ik het kon aanpassen, maar dat werd best veel werk, waarvan ik niet precies wist hoe dat te doen, zonder alles te breken.
Misschien dit als een "future improvement" meenemen?

Reply:
Goed punt, we moeten ook niet alles gelijk proberen aan te passen ;)